mono/tests/dim-implements-multiple-interfaces.il

```diff
@@ -0,0 +1,196 @@
+	IL_000b:  nop 
+	IL_000c:  ret 
+    } // end of method I2::Application.I1.M1
+
+  } // end of class Application.I2
+}
+
+.namespace Application
+{
+  .class interface public auto ansi abstract I3
+  	implements Application.I1  {
+
+    // method line 3
+    .method public final virtual hidebysig 
+           instance default void Application.I1.M1 ()  cil managed 
+    {
+        // Method begins at RVA 0x206c
+	.override class Application.I1::M1
+	// Code size 13 (0xd)
+	.maxstack 8
+	IL_0000:  nop 
+	IL_0001:  ldstr "I3.I1.M1"
+	IL_0006:  call void class [mscorlib]System.Console::WriteLine(string)
+	IL_000b:  nop 
+	IL_000c:  ret 
+    } // end of method I3::Application.I1.M1
+
+  } // end of class Application.I3
+}
+
+.namespace Application
+{
+  .class interface public auto ansi abstract I4
+  	implements Application.I1  {
+
+    // method line 4
+    .method public final virtual hidebysig 
+           instance default void Application.I1.M1 ()  cil managed 
+    {
+        // Method begins at RVA 0x207a
+	.override class Application.I1::M1
+	// Code size 13 (0xd)
+	.maxstack 8
+	IL_0000:  nop 
+	IL_0001:  ldstr "I4.I1.M1"
+	IL_0006:  call void class [mscorlib]System.Console::WriteLine(string)
+	IL_000b:  nop 
+	IL_000c:  ret 
+    } // end of method I4::Application.I1.M1
+
+  } // end of class Application.I4
+}
+
+.namespace Application
+{
+  .class private auto ansi beforefieldinit Test10
+  	extends [mscorlib]System.Object
+  	implements Application.I1, Application.I2, Application.I3, Application.I4  {
+
+    // method line 5
+    .method public final virtual hidebysig newslot 
+           instance default void M1 ()  cil managed 
+    {
+        // Method begins at RVA 0x2088
+	// Code size 13 (0xd)
+	.maxstack 8
+	IL_0000:  nop 
+	IL_0001:  ldstr "Test10.M1"
+	IL_0006:  call void class [mscorlib]System.Console::WriteLine(string)
+	IL_000b:  nop 
+	IL_000c:  ret 
+    } // end of method Test10::M1
+
+    // method line 6
+    .method public hidebysig specialname rtspecialname 
+           instance default void '.ctor' ()  cil managed 
+    {
+        // Method begins at RVA 0x2096
+	// Code size 8 (0x8)
+	.maxstack 8
+	IL_0000:  ldarg.0 
+	IL_0001:  call instance void object::'.ctor'()
+	IL_0006:  nop 
+	IL_0007:  ret 
+    } // end of method Test10::.ctor
+
+  } // end of class Application.Test10
+}
+
+.namespace Application
+{
+  .class private auto ansi beforefieldinit MainClass
+  	extends [mscorlib]System.Object
+  {
+
+    // method line 7
+    .method public static hidebysig 
+           default void Main (string[] args)  cil managed 
+    {
+        // Method begins at RVA 0x20a0
+	.entrypoint
+	// Code size 15 (0xf)
+	.maxstack 1
+	.locals init (
+		class Application.I1	V_0)
+	IL_0000:  nop 
+	IL_0001:  newobj instance void class Application.Test10::'.ctor'()
+	IL_0006:  stloc.0 
+	IL_0007:  ldloc.0 
+	IL_0008:  callvirt instance void class Application.I1::M1()
+	IL_000d:  nop 
+	IL_000e:  ret 
```

Comment:
@thaystg I think we should improve the test.  Right now all that we're testing is that Mono doesn't crash when one of these DIM methods is called.

We should change the test so that we're verifying that the _expected_ method is being called.
One way to do that is to change all the implementations of `I1.M1`  that we don't want to run to throw an exception.  That way mono will exit with a non-zero exit code if they ever get called.

Also there's a merge conflict that needs to be resolved.

Reply:
Done!